app/models/apple/publisher.rb

```diff
@@ -106,20 +106,30 @@ def poll!(eps = episodes_to_sync)
       end
     end
 
-    def publish!(eps = episodes_to_sync)
```

Comment:
Removing this default parameter fixes up the caching of episode state. So when we un-archive the episode, the subsequent call to `episodes_to_sync` then freshly filters episode state (derived from the private feed).